src/main/java/uk/gov/companieshouse/transactions/web/service/transaction/impl/TransactionsServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.companieshouse.transactions.web.service.transaction.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.ApiClient;
+import uk.gov.companieshouse.api.error.ApiErrorResponseException;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+import uk.gov.companieshouse.transactions.web.api.ApiClientService;
+import uk.gov.companieshouse.transactions.web.exception.ServiceException;
+import uk.gov.companieshouse.transactions.web.service.transaction.TransactionsService;
+
+@Service
+public class TransactionsServiceImpl implements TransactionsService {
+
+    @Autowired
+    ApiClientService apiClientService;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Transaction getTransaction(String transactionId) throws ServiceException {
+
+        ApiClient apiClient = apiClientService.getApiClient();
+
+        try {
+            return apiClient.transaction(transactionId).get();
+        } catch (ApiErrorResponseException e) {
+            throw new ServiceException("Error retrieving transaction", e);
```

Comment:
are we logging it in where this is caught?

Reply:
Yeah, this will be logged at the controller

Reply:
:+1: